src/webui/webapplication.cpp

```diff
@@ -373,6 +373,9 @@ void WebApplication::doProcessRequest()
                 }
             }
             break;
+        case QMetaType::UnknownType:
```

Comment:
what will trigger this case?

Reply:
> what will trigger this case?

Response w/o data.

Reply:
Should all responses with no data use `204`? Shouldn't it depend on what was requested?

Reply:
For example, `setPreferencesAction()` don't return data but it is expected for the client to receive `204 No Content` in the response? Shouldn't the client expect it to be `200 OK`?
I mean `204` should probably not be sent unconditionally and only some api should use it.

Reply:
Any request that doesn't result in data being sent back to the client should receive a 204. This page has useful information: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status.

Reply:
> I mean `204` probably should not be sent unconditionally and only some api should use it.

What's the point of it then?
What's the point of it then?
IIRC, it means the same as 200 but give the client a hint of that request processing can be optimized.


> Shouldn't the client expect it to be `200 OK`?

It should expect the code that is declared by API.

Reply:
@Piccirello 
What is the motivation behind this change?

Reply:
> This change is part of a broader push to have the WebAPI send back semantically correct status codes.

I also now make use of 202 in another PR. We currently have APIs, like auth, that send back a 200 in the case of a failure.